sys/include/net/gnrc/netif.h

```diff
@@ -187,6 +187,41 @@ typedef struct {
     kernel_pid_t pid;                       /**< PID of the network interface's thread */
 } gnrc_netif_t;
 
+
+/**
+ * @brief   Check if the device belonging to the given netif uses the legacy
+ *          netdev API
+ *
+ * netdevs using the legacy API have to depend on the (pseudo-)module
+ * netdev_legaqcy_api, netdevs using the new API have to depend on the
+ * (pseudo-)module netdev_new_api. If only one of the pseudo modules is used,
+ * this function can be constant folded. For boards mixing legacy and new API
+ * netdevs, this will check the flavor at runtime.
+ */
+static inline bool gnrc_netif_netdev_legacy_api(gnrc_netif_t *netif)
+{
+    static_assert(IS_USED(MODULE_NETDEV_NEW_API) || IS_USED(MODULE_NETDEV_LEGACY_API),
+                  "used netdev misses dependency to netdev_legacy_api");
+    if (!IS_USED(MODULE_NETDEV_NEW_API)) {
+        return true;
+    }
+    if (!IS_USED(MODULE_NETDEV_LEGACY_API)) {
+        return false;
+    }
+    return (netif->dev->driver->confirm_send == NULL);
+}
+
+/**
+ * @brief   Check if the device belonging to the given netif uses the new
+ *          netdev API
```

Comment:
Is there anywhere documented what the 'new netdev API' is and what it takes to convert to it?

Reply:
It is documented in `netdev_driver_t`. The difference is twofold: The old API has a blocking send function, while the new send function should return when the frame to send is loaded into the devices buffer and TX ia triggered. The new netdevs can rely that ISRs can still be handled and no further call to send is done until TX_COMPLETED was signaled. Secondly, the new API has the confirm send function to fetch the result of the transmission (success / error and optionally TX stats such as L2 ACKs/NACKs, TX timestamp, ...).

My plan is to migrate all drivers quickly and drop all references two new/old API afterwards.

Reply:
I added a reference to `netdev_driver_t::confirm_send` to the doc.